app/models/user.rb

```diff
@@ -292,6 +292,21 @@ def self.where_case_insensitive(field, val)
     User.where("lower(#{field}) = ?", val.respond_to?(:downcase) ? val.downcase : val.to_s)
   end
 
+  # Get the current User
+  # Useful for accessing current User info in views
+  # @return [User] the current logged in user
+  def self.current
+    Thread.current[:user]
+  end
+
+  # Set the current User
+  # Used to set the current User from current_user in controllers
+  # @param user user to set
+  # @return [User] the current logged in user
+  def self.current=(user)
+    Thread.current[:user] = user
```

Comment:
don't use the local thread storage unless is really necessary and for this case we have already current_user in place...